pkg/steps/source.go

```diff
@@ -690,7 +693,32 @@ func getReasonsForUnreadyContainers(p *coreapi.Pod) string {
 			reason = "unknown"
 			message = "unknown"
 		}
-		_, _ = builder.WriteString(fmt.Sprintf("\n* Container %s is not ready with reason %s and message %s", c.Name, reason, message))
+		if message != "" {
+			message = fmt.Sprintf(" and message %s", message)
+		}
+		_, _ = builder.WriteString(fmt.Sprintf("\n* Container %s is not ready with reason %s%s", c.Name, reason, message))
+	}
+	return builder.String()
+}
+
+func getEventsForPod(pod *coreapi.Pod, client coreclientset.EventInterface, ctx context.Context) string {
```

Comment:
I don't see how this is related to the `source` step -- it's only called from `template.go` so I think it should be in that file.

Reply:
That's where all of our helpers are for this stuff, if we want to clean it up we can do that separately?